src/Profiles.ts

```diff
@@ -61,45 +61,17 @@ export class Profiles { // Processing stops if there are no profiles detected
         throw new Error(localize("loadNamedProfile.error.profileName", "Could not find profile named: ")
             + name + localize("loadNamedProfile.error.period", "."));
     }
+
     public getDefaultProfile(): IProfileLoaded {
         return this.defaultProfile;
     }
-    public async refresh() {
-        if (this.isSpawnReqd() === 0) {
-            this.allProfiles = ProfileLoader.loadAllProfiles();
-            try {
-                this.defaultProfile = ProfileLoader.loadDefaultProfile(this.log);
-            } catch (err) {
-                // Unable to load a default profile
-                this.log.warn(localize("loadNamedProfile.warn.noDefaultProfile",
-                    "Unable to locate a default profile. CLI may not be installed. ") + err.message);
-            }
-        } else {
-            const profileManager = new CliProfileManager({
-                profileRootDirectory: path.join(os.homedir(), ".zowe", "profiles"),
-                type: "zosmf"
-            });
-            this.allProfiles = (await profileManager.loadAll()).filter((profile) => {
-                return profile.type === "zosmf";
-            });
-            if (this.allProfiles.length > 0) {
-                this.defaultProfile = (await profileManager.load({ loadDefault: true }));
-            } else {
-                ProfileLoader.loadDefaultProfile(this.log);
-            }
-        }
-    }
 
-    public listProfile() {
-        try {
-            this.allProfiles = ProfileLoader.loadAllProfiles();
-        } catch (error) {
-            vscode.window.showErrorMessage(error.message);
-        }
-        this.allProfiles.map((profile) => {
-            return profile.name;
+    public async refresh() {
+        const profileManager = await this.getCliProfileManager("zosmf");
+        this.allProfiles = (await profileManager.loadAll()).filter((profile) => {
+            return profile.type === "zosmf";
```

Comment:
We may want to parameterize the profile type in order to make it easier in case we need to expose it in the settings, (FTP and SSH in the future?)

Reply:
Yes I thought we would probably end up going that route at some point but thought we would simply re-engineer then.